src/mumble/Database.cpp

```diff
@@ -396,30 +399,40 @@ void Database::setLocalMuted(const QString &hash, bool muted) {
 	execQueryAndLogFailure(query);
 }
 
-bool Database::isChannelFiltered(const QByteArray &server_cert_digest, const int channel_id) {
+ChannelFilter Database::getChannelFilterMode(const QByteArray &server_cert_digest, const int channel_id) {
 	QSqlQuery query(db);
 
 	query.prepare(QLatin1String(
-		"SELECT `channel_id` FROM `filtered_channels` WHERE `server_cert_digest` = ? AND `channel_id` = ?"));
+		"SELECT `filter_mode` FROM `filtered_channels` WHERE `server_cert_digest` = ? AND `channel_id` = ?"));
 	query.addBindValue(server_cert_digest);
 	query.addBindValue(channel_id);
 	execQueryAndLogFailure(query);
 
-	return query.next();
+	if (query.first()) {
+		return static_cast< ChannelFilter >(query.value(0).toInt());
+	}
+
+	return ChannelFilter::NORMAL;
 }
 
-void Database::setChannelFiltered(const QByteArray &server_cert_digest, const int channel_id, const bool hidden) {
+void Database::setChannelFilterMode(const QByteArray &server_cert_digest, const int channel_id,
+									const ChannelFilter filterMode) {
 	QSqlQuery query(db);
 
-	if (hidden)
-		query.prepare(
-			QLatin1String("INSERT INTO `filtered_channels` (`server_cert_digest`, `channel_id`) VALUES (?, ?)"));
-	else
-		query.prepare(
-			QLatin1String("DELETE FROM `filtered_channels` WHERE `server_cert_digest` = ? AND `channel_id` = ?"));
+	switch (filterMode) {
+		case ChannelFilter::NORMAL:
+			query.prepare(
+				QLatin1String("DELETE FROM `filtered_channels` WHERE `server_cert_digest` = ? AND `channel_id` = ?"));
+			break;
+		default:
```

Comment:
Please replace default clause with explicit case labels (with fall-through) - again to ensure compiler warnings are generated if there ever is an additional, unhandled filter mode